src/gt4py/next/ffront/foast_passes/type_deduction.py

```diff
@@ -620,6 +620,19 @@ def _deduce_binop_type(
         right: foast.Expr,
         **kwargs: Any,
     ) -> Optional[ts.TypeSpec]:
+        if (
+            isinstance(left.type, ts.DimensionType)
+            and isinstance(right.type, ts.ScalarType)
+            and type_info.is_integral(right.type)
+        ):
+            return ts.OffsetType(source=left.type.dim, target=(left.type.dim,))
+        if (
```

Comment:
Why is this cases need? Do we want something like `Ioff+1` which I assume is what this is covering?

Reply:
This is for IDim+1+2, which we would like to support for embedded. I added the respective case to _visit_shift in foast_to_itir.py again as one otherwise gets the "Unexpected shift arguments!" error.

Reply:
Makes sense. I wondered how this actually works now in embedded since I don't see an `__add__` in `FieldOffset`.

Reply:
The `__add__` in CartesianConnectivity takes care of that.